frontend/src/components/ScheduledEventsTable/ScheduledEventsTable.tsx

```diff
@@ -1,29 +1,39 @@
-import React, { FunctionComponent, ReactNode } from 'react';
+import classNames from 'classnames';
+import React, { FunctionComponent, ReactNode, useState } from 'react';
 import { Table } from 'react-bootstrap';
 import { ScheduledEvent } from '../../types/scheduledEvents';
 import { ScheduledEventsTableRow } from '../ScheduledEventsTableRow';
-
 export interface ScheduledEventTableProps {
   currentPage: number;
   pageLimit: number;
   events: Array<{}>;
+  handleRunHistory: (id: number, name: string) => void;
 }
 export const ScheduledEventsTable: FunctionComponent<ScheduledEventTableProps> = (props) => {
-  const offset = (props.currentPage - 1) * props.pageLimit;
+  const [activeRow, setActiveRow] = useState(0);
+
+  const handleClick = (id: number, name: string) => (): void => {
+    if (id !== activeRow) {
+      setActiveRow(id);
+      props.handleRunHistory(id, name);
+    }
+  };
 
   const renderRows = (): ReactNode =>
-    props.events.map((event: ScheduledEvent, index: number) => {
+    props.events.map((event: ScheduledEvent) => {
       return (
         <ScheduledEventsTableRow
-          key={index}
-          id={offset + index + 1}
+          key={event.id}
+          id={event.id}
           name={event.name}
           description={event.description || ''}
           enabled={event.enabled}
           interval={event.interval}
           interval_type={event.interval_type}
           repeat={event.repeat}
           start_date={event.start_date}
```

Comment:
I probably missed this in the previous scheduled events PR review ... mentioned in for the history though ... instead of passing each property as its own prop, please pass in the entire event. A lot cleaner.